.github/workflows/main.yml

```diff
@@ -1,40 +1,151 @@
+
+          - os: macos-latest
+            bazel-cache-path: /private/var/tmp/_bazel_*
+            build-command: |
+              bazel build //zetasql/local_service:libremote_server.dylib --features=-supports_dynamic_linker
+              install_name_tool -id "@loader_path/libremote_server.dylib" bazel-bin/zetasql/local_service/libremote_server.dylib
+            lib-path: bazel-bin/zetasql/local_service/libremote_server.dylib
+            artifacts-archive: libremote_server.dylib.tar.gz
+
+    outputs:
+      bazel-version: ${{ steps.set-bazel-version.outputs.bazel-version }}
+
+    runs-on: ${{ matrix.os }}
 
-    # Steps represent a sequence of tasks that will be executed as part of the job
     steps:
-      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
-      - uses: actions/checkout@v2
-          
-      - name: Install Bazel
-        uses: abhinavsingh/setup-bazel@v3
+      - name: Checkout
+        uses: actions/checkout@v3
+
+      - name: Set environment variable with bazel version
+        id: set-bazel-version
+        run: |
+          echo "::set-output name=bazel-version::$(cat .bazelversion)"
+          echo $(cat .bazelversion)
+
+      - name: Setup bazelisk
+        uses: bazelbuild/setup-bazelisk@v2
+    
+      - name: Mount bazel cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ${{ matrix.bazel-cache-path }}
+            ./bazel-bin
+            ./bazel-out
+            ./bazel-snowsql
+          key: ${{ runner.os }}-${{ steps.set-bazel-version.outputs.bazel-version }}-bazel
+
+      - name: Bazel build
+        run: ${{ matrix.build-command }}
+
+      - name: Archive build artifacts
+        run: tar -cvz -f ${{ matrix.artifacts-archive }} ${{ matrix.lib-path }}
+
+      - name: Upload Artifacts
+        uses: actions/upload-artifact@v3
+        with:
+          name: ${{ runner.os }}
+          path: ${{ matrix.artifacts-archive }}
+
+  test:
+    needs: build
+
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - os: ubuntu-latest
+            bazel-cache-path: ~/.cache/bazel
+            lib-path: bazel-bin/zetasql/local_service/libremote_server.so
+            test-command: |
+              bazel test //zetasql/analyzer/...
+              bazel test //zetasql/base/...
+              bazel test //zetasql/common/...
+              bazel test //zetasql/compliance/...
+              bazel test //zetasql/local_service/...
+              bazel test //zetasql/parser/...
+
+          - os: macos-latest
+            bazel-cache-path: /private/var/tmp/_bazel_*
+            lib-path: bazel-bin/zetasql/local_service/libremote_server.dylib
+            test-command: | # Temporary skip 2 tests for MacOS
+              bazel test --features=-supports_dynamic_linker -- //zetasql/analyzer/... -//zetasql/analyzer:analyzer_set_operation_test -//zetasql/analyzer:analyzer_test
```

Comment:
Is there any issue with approach where we run ALL test but without these two (or more)? Something like:
`bazel test --features=-supports_dynamic_linker -- ... -//zetasql/analyzer:analyzer_set_operation_test -//zetasql/analyzer:analyzer_test`

Reply:
Will be considered in another PR.